compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IdSignature.kt

```diff
@@ -88,6 +88,9 @@ sealed class IdSignature {
             other is PublicSignature && packageFqName == other.packageFqName && declarationFqName == other.declarationFqName &&
                     id == other.id && mask == other.mask
 
+        fun equalsWithNames(other: Any?): Boolean =
```

Comment:
Why do we need this function? Why doesn't normal `equals` work?

Reply:
I made this because faced a problem with KClass in cases of annotations. Mask in backend usage for it equals 1, but for FIR - 0. I didn't debug deeply in FIR and just made this hack. If  you have an idea how to fix this it'd be cool.

Reply:
Looks like it's the same problem which you worked around by keeping the old FqName-based `isClassType`?

Just replace `isKClass` with
```
fun IrType.isKClass(): Boolean = isClassType(StandardNames.FqNames.kClass, false)
```

and then we can use normal `IdSignature.equals` (at least all JVM IR and all FIR tests pass for me)